jwst/stpipe/utilities.py

```diff
@@ -211,3 +212,23 @@ def query_step_status(datamodel, cal_step):
         return getattr(datamodel[0].meta.cal_step, cal_step, NOT_SET)
     else:
         return getattr(datamodel.meta.cal_step, cal_step, NOT_SET)
+
+
+def invariant_filename(save_model_func):
```

Comment:
Not crucial, but it would be helpful to write a unit test for this function.  I think it could be pretty simple.  Writing some pseudocode on-the-fly, I'm thinking something like
```
def mock_save_func(model):
    model.meta.filename = "changed"
    return model

def test_invariant_filename(input_model):
    input_model.meta.filename = "initial_name"
    invariant_save_func = invariant_filename(mock_save_func)
    output_model = invariant_save_func(input_model)
    assert output_model.meta.filename == "initial_name"
```